src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -274,10 +297,17 @@ private T createOrUpdate(Request req, Response res) {
      * Get entity ID from request.
      */
     private String getIdFromRequest(Request req) {
-        String id = req.params("id");
-        if (id == null) {
-            logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "Must provide id");
+        return getParamFromRequest(req, "id");
+    }
+
+    /**
+     * Get entity attribute value from request.
+     */
+    private String getParamFromRequest(Request req, String paramName) {
```

Comment:
Can we rename this to `getRequiredParamFromRequest` and add a note that this method will halt the request if the param is not provided?